lms/djangoapps/certificates/tests/test_services.py

```diff
@@ -61,3 +66,27 @@ def test_invalidate_certificate(self):
                 'status': CertificateStatuses.unavailable
             }
         )
+
+    @override_waffle_flag(CERTIFICATES_USE_ALLOWLIST, active=True)
+    def test_invalidate_certificate_allowlist(self):
+        """
+        Verify that CertificateService does not invalidate the certificate if it is allowlisted
+        """
+        u = UserFactory.create()
+        c = CourseFactory()
+        course_key = c.id  # pylint: disable=no-member
+        GeneratedCertificateFactory.create(
+            status=CertificateStatuses.downloadable,
+            user=u,
+            course_id=course_key,
+            grade=1.0
+        )
+        CertificateWhitelistFactory(
+            user=u,
+            course_id=course_key
+        )
+        success = self.service.invalidate_certificate(u.id, course_key)
+        self.assertFalse(success)
+
+        cert = GeneratedCertificate.objects.get(user=u.id, course_id=course_key)
+        self.assertTrue(cert.status, CertificateStatuses.unavailable)
```

Comment:
I feel like I may be missing something here...

I expected a comparison of the `cert.status` to `CertificateStatuses.downloadable`? Shouldn't the invalidation be skipped if the user is on the allowlist?

Reply:
Heh. I used the wrong assert (should have been equals). Thank you for catching this!